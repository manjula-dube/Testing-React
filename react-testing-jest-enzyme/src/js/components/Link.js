import React, { Component } from "react";

export default class Link extends Component {
  constructor(props) {
    super(props);
    this.state = { clicked: false };
  }

  handleClick() {
    alert("clicked");
    this.setState({ clicked: true });
  }

  render() {
    const { title, url } = this.props;
    return (
      <div className="link-container">
        <a href={url} target="_blank" onClick={() => this.handleClick()}>
          {title}
        </a>
      </div>
    );
  }
}
