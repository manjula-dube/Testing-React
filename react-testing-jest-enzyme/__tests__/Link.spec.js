
 import React from 'react';
 import { shallow } from 'enzyme';
 import { shallowToJson } from 'enzyme-to-json';
 import Link from '../src/js/components/Link'

 describe('Link', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Link title="mockTitle" url="mockUrl" />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
 });

 it('should handle state changes', () => {
  const output = shallow(
    <Link title="mockTitle" url="mockUrl" />
  );
  
  expect(output.state().clicked).toEqual(false);
  output.simulate('click');
  expect(output.state().clicked).toEqual(false);
});

it('should handle the click event', () => {
  window.alert = jest.fn();
  const output = shallow(
    <Link title="mockTitle" url="mockUrl" />
  );
  
  output.find('a').simulate('click');
  expect(window.alert).toHaveBeenCalledWith('clicked');
});
 


 



