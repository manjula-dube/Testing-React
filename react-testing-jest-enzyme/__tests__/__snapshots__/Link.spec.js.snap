exports[`Link should render correctly 1`] = `
<div
  className="link-container">
  <a
    href="mockUrl"
    onClick={[Function]}
    target="_blank">
    mockTitle
  </a>
</div>
`;
